test/test_torch.py

```diff
@@ -2620,6 +2620,26 @@ def test_index_add(self):
             dest2[idx[i]] = dest2[idx[i]] + src[i]
         self.assertEqual(dest, dest2)
 
+    # add coverage for issue with atomic add that appeared only for 
+    # specific dtypes on cuda:
+    # https://github.com/pytorch/pytorch/issues/29153
+    def test_dtypes_for_index_add(self):
+        def is_signed(dtype):
+            return torch.is_signed(torch.tensor([], dtype=dtype))
+        for device in torch.testing.get_all_device_types():
+            for dtype in torch.testing.get_all_math_dtypes(device):
+                size = [5, 5]
+                if dtype.is_floating_point:
+                    tensor = torch.rand(size, dtype=dtype, device=device)
+                elif is_signed(dtype):
+                    tensor = torch.randint(-5, 15, size, dtype=dtype, device=device)
+                else:
+                    tensor = torch.randint(0, 10, size, dtype=dtype, device=device)
+                # index_add calls atomicAdd on cuda.
+                empty = torch.zeros(size, dtype=dtype, device=device)
```

Comment:
nit: this isn't an "empty" tensor, it is zero-filled so it is better to call it something like "zeros"